RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -428,39 +621,53 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
   // get total charge in calibration channels
   if(hCalib.isValid() == true)
   {
+
+
+     std::vector<std::vector<int> > lasmon_adcs(laserMonCBoxList_.size(), std::vector<int>());
+     std::vector<std::vector<int> > lasmon_capids(laserMonCBoxList_.size(), std::vector<int>());
+
      for(HcalCalibDigiCollection::const_iterator digi = hCalib->begin(); digi != hCalib->end(); digi++)
      {
         if(digi->id().hcalSubdet() == 0)
            continue;
 
-        /*
-        HcalCalibDetId cell = digi->id();
-        DetId detcell = (DetId)cell;
-        
-        const HcalChannelStatus* mydigistatus=myqual->getValues(detcell.rawId());
-
-        if(mySeverity->dropChannel(mydigistatus->getValue()))
-           continue;
-        if(digi->zsMarkAndPass())
-           continue;
-
-        const HcalQIECoder *channelCoder = conditions->getHcalCoder(cell);
-	const HcalQIEShape* shape = conditions->getHcalShape(channelCoder);
-        HcalCoderDb coder(*channelCoder, *shape);
-
-        CaloSamples tool;
-        coder.adc2fC(*digi, tool);
-
-        for(int i = 0; i < (int)digi->size(); i++)
-           TotalCalibCharge = TotalCalibCharge + tool[i];
-        */
-
-	// Original code computes total calib charge over all digis.  While I think it would be more useful to skip
-	// zs mark-and-pass channels, I keep this computation as is.  Individual HBHE and HF variables do skip
-	// the m-p channels.  -- Jeff Temple, 6 December 2012
-
-	for(int i = 0; i < (int)digi->size(); i++)
-	  TotalCalibCharge = TotalCalibCharge + adc2fC[digi->sample(i).adc()&0xff];
+        // Fill the lasermonitor channels
+        if( fillLaserMonitor_ ) {
+          int cboxch  = digi->id().cboxChannel( );
+          int iphi    = digi->id().iphi();
+          int ieta    = digi->id().ieta();
+         
+          // check that we have the correct cboxch
+          if (std::find( laserMonCBoxList_.begin(), laserMonCBoxList_.end(),
+                         cboxch ) != laserMonCBoxList_.end() ) {
+              // check that we have a contained IPhi
+            if( std::find( laserMonIPhiList_.begin(), laserMonIPhiList_.end(),
+                           iphi ) != laserMonIPhiList_.end() ) {
+              // check that we have a contained IEta
+              if( std::find( laserMonIEtaList_.begin(), laserMonIEtaList_.end(),
+                             ieta ) != laserMonIEtaList_.end() ) {
+                // we have a lasmon channel, find the index in the list of inputs
+                for( unsigned idx = 0; idx < laserMonCBoxList_.size(); ++idx ) {
```

Comment:
hi @jkunkle - sorry, I thought I had commented on this before, but it seems not... these finds do not seem to be needed given the loop that follows here. The check in the loop takes advantage of the knowledge that the index is always the same.

Reply:
Hi @davidlange6 .  The point here is to only run the loop for the 4-8 channels of interest rather than for every calibration channel.  These std::finds could easily be removed, but the loop will run for every channel.  Alternatively one could just keep the cBoxList check, which would be sufficient to filter out these channels.  What do you think?